pytorch_lightning/core/lightning.py

```diff
@@ -171,6 +171,12 @@ def automatic_optimization(self) -> bool:
         """
         return True
 
+    def _prepare_batch_for_transfer(self, batch: Any, device: Optional[torch.device] = None):
```

Comment:
Hey @rohitgr7,

Can we find a better name `_prepare_batch_for_transfer` ? Something more explicit like: `_batch_transfer_and_pre_post_transforms`

T.C

Reply:
applying transformations is just one use case for these hooks. So I thought not to include `transforms` in the function name.

Reply:
Hey @rohitgr7, 

I know you already invested lot of time in this PR but I find `prepare` slightly confusing as it refers to `pre_{something}`.

Something like `_batch_transfer_procedure` or `_batch_transfer_handler` might explain better the process. 

Best,
T.C